cmd/internal/datastore/pq/transactions.go

```diff
@@ -0,0 +1,140 @@
+package pq
+
+import (
+	"context"
+	"math/big"
+
+	"github.com/Masterminds/squirrel"
+	"github.com/jmoiron/sqlx"
+	"github.com/mailchain/mailchain/cmd/internal/datastore"
+	"github.com/pkg/errors"
+)
+
+type TransactionStore struct {
+	db *sqlx.DB
+}
+
+type transaction struct {
+	Protocol uint8  `db:"protocol"`
+	Network  uint8  `db:"network"`
+	Hash     []byte `db:"hash"`
+
+	From      []byte `db:"tx_from"`
+	To        []byte `db:"tx_to"`
+	Data      []byte `db:"tx_data"`
+	BlockHash []byte `db:"tx_block_hash"`
+	Value     []byte `db:"tx_value"`
+	GasUsed   []byte `db:"tx_gas_used"`
+	GasPrice  []byte `db:"tx_gas_price"`
+}
+
+func NewTransactionStore(db *sqlx.DB) (datastore.TransactionStore, error) {
+	return &TransactionStore{db: db}, nil
+}
+
+func (s *TransactionStore) PutTransaction(ctx context.Context, protocol, network string, hash []byte, tx *datastore.Transaction) error {
+	p, n, err := getProtocolNetworkUint8(protocol, network)
+	if err != nil {
+		return errors.WithStack(err)
+	}
+
+	sql, args, err := squirrel.Insert("transactions").
+		Columns("protocol", "network", "hash", "tx_from", "tx_to", "tx_data", "tx_block_hash", "tx_value", "tx_gas_used", "tx_gas_price").
+		Values(p, n, hash, tx.From, tx.To, tx.Data, tx.BlockHash, tx.Value, tx.GasUsed, tx.GasPrice).
+		PlaceholderFormat(squirrel.Dollar).
+		Suffix("ON CONFLICT DO UPDATE SET tx_from = $, tx_to = $, tx_data = $, tx_block_hash = $, tx_value = $, tx_gas_used = $, tx_gas_price = $",
+			tx.From, tx.To, tx.Data, tx.BlockHash, tx.Value, tx.GasUsed, tx.GasPrice).
+		ToSql()
+
+	if err != nil {
+		return errors.WithStack(err)
+	}
+
+	if _, err := s.db.Exec(sql, args...); err != nil {
+		return errors.WithStack(err)
+	}
+
+	return nil
+}
+
+func (s *TransactionStore) GetTransactionsFrom(ctx context.Context, protocol, network string, address []byte) ([]datastore.Transaction, error) {
+	p, n, err := getProtocolNetworkUint8(protocol, network)
+	if err != nil {
+		return nil, errors.WithStack(err)
+	}
+
+	sql, args, err := squirrel.Select("tx_from", "tx_to", "tx_data", "tx_block_hash", "tx_value", "tx_gas_used", "tx_gas_price").
+		From("transactions").
+		PlaceholderFormat(squirrel.Dollar).
+		Where(squirrel.Eq{"protocol": p}).
+		Where(squirrel.Eq{"network": n}).
+		Where(squirrel.Eq{"tx_from": address}).
```

Comment:
There is a lot of duplication in `GetTransactionsFrom` and `GetTransactionsTo` methods, I think it could be simplified with an internal method that changes this line